model/plugin_kvset_options.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+// PluginKVSetOptions contains information on how to store a value in the plugin KV store.
+type PluginKVSetOptions struct {
+	EncodeJSON      bool        // If true store the information encoded as JSON
```

Comment:
I'm not sure what EncodeJSON means... And the comment doesn't help because it makes me think you are trying to "store the information encoded as JSON" (which is not really true, right?). So maybe something like:
```suggestion
	isJSON                bool        // If true, we are trying to write a JSON object to the key value store
```
This is similar to the way @lieut-data wrote the docs for `KVSetJSON`, which I find very clear:
```
// KVSetJSON is a wrapper around KVSet to simplify writing a JSON object to the key value store.
```

Reply:
If `EncodeJSON` is set, we technical call `json.Marshal` before we store it. https://github.com/mattermost/mattermost-server/pull/11818/files#diff-e715a568b53768e9be119fabbb11ed54R63-R69

Doesn't this fit "store the information encoded as JSON"?

Reply:
Okay, I just found the english a bit confusing. I would find it less confusing if we had something like: 
```suggestion
	EncodeJSON      bool        // If true, store the JSON encoding of newValue
```

This matches the godoc for marshal, which makes sense to me.